src/graphs/BasicGraphs.jl

```diff
@@ -159,6 +159,28 @@ function add_edges!(g::HasGraph, srcs::AbstractVector{Int},
   add_parts!(g, :E, n, (src=srcs, tgt=tgts, kw...))
 end
 
+""" Add reflexive edges to a graph
+
+Setting the keyword `idempotent::Bool=true` will skip adding reflexive edges to vertices which already have them
+"""
+function add_reflexives!(g::HasGraph; idempotent::Bool=false)
+    if idempotent
+        out = Int[]
+        for v in vertices(g)
+            if isempty(incident(g, v, :src) ∩ incident(g, v, :tgt))
+                e′ = add_edge!(g, v, v)
```

Comment:
to be consistent with the rest of this file, these tabs should be just two spaces